### PR TITLE
[vSphere] Apply doc guidelines and general editing

### PR DESCRIPTION
**Pull request by alaudazzi**
## Proposed commit message

This PR:
- applies the doc template
- removes duplications
- proposes fixes as per Elastic general writing guidelines

Closes https://github.com/elastic/integration-docs/issues/557

### CODE REVIEW

**Review by shmsr — APPROVED** [review]

File: packages/vsphere/_dev/build/docs/README.md [hunk 1]
```diff
@@ -16,143 +16,149 @@ The vSphere integration collects logs and metrics.
...
+- **`host`**: Collects host metrics from VMware vSphere, including performance statistics such as CPU usage, memory usage, disk I/O, and network activity.
+- **`resourcepool`**: Collects metrics from VMware vSphere, such as CPU and memory usage, CPU and memory reservation, and CPU and memory limit.
+- **`virtualmachine`**: Gathers virtual machine metrics from VMware vSphere, including performance statistics such as status, uptime, CPU usage, memory usage, and network activity.
+- **`network`**: Gathers metrics and status information related to VMware vSphere networks, including network accessibility, connected hosts and virtual machines, configuration health, and network type.
+
+Note:
+
+Users can monitor and check the log inside the ingested documents for vSphere in the `logs-*` index pattern from `Discover`, and for metrics, the index pattern is `metrics-*`.
+
+## Supported periods
```
Comment (ishleenk17) [inline packages/vsphere/_dev/build/docs/README.md:36]: @alaudazzi : The supported periods is mor of a config related information, hence it was kept under notes.
Not sure if giving it as a separate heading is a good idea. WDYT ?

Comment (ishleenk17) [reply packages/vsphere/_dev/build/docs/README.md:36]: Although I see you have done this to use its reference towards end of the documentation

Comment (alaudazzi) [reply packages/vsphere/_dev/build/docs/README.md:36]: @ishleenk17 A separate heading enhances visibility on an important piece of content that is also referenced later in the document.

**Pull request merged**